### PR TITLE
cmd: support OLLAMA_CLIENT_HOST environment variable

### PR DESCRIPTION
**Pull request by bmizerany**
This commit adds support for the OLLAMA_CLIENT_HOST environment variable. This variable can be used to specify the host to which the client should connect. This is useful when the client is running somewhere other than the host where the server is running.

The new api.FromEnv function is used to configure clients from the environment. Clients wishing to use the environment variable being consistent with the Ollama CLI can use this new function.

### CODE REVIEW

**Comment by drhino** [comment]
This is already implemented in:
https://github.com/jmorganca/ollama/blob/main/cmd/cmd.go#L517

**Comment by drhino** [comment]
Only noticed #261 now
Maybe you want to update this file too:
https://github.com/jmorganca/ollama/blob/main/server/routes.go#L308

**Comment by bmizerany** [comment]
That is a change required for any client to work, or just browsers?

If only the latter, maybe that is for another PR?

**Comment by drhino** [comment]
I think thats only for the browser.. But it would be great to have!

**Comment by bmizerany** [comment]
Agreed. Maybe that's for another PR? I'm not seeing docs mention browser support atm, which leads me to think this isn't the right PR for browser support.

File: api/client.go [hunk 1]
```diff
@@ -33,14 +40,34 @@ func checkError(resp *http.Response, body []byte) error {
 	return apiError
 }
 
+// Host returns the default host to use for the client. It is determined in the following order:
+// 1. The OLLAMA_CLIENT_HOST environment variable
+// 2. The default host (localhost:11434)
+func Host() string {
```
Comment (justinmayer) [inline api/client.go:46]: I think `Host()` is a bit too generic. What if other host-related functions are added later? I think `GetHost()` would be more specific and descriptive.

File: api/client.go [hunk 2]
```diff
@@ -33,14 +40,34 @@ func checkError(resp *http.Response, body []byte) error {
...
+	if envHost != "" {
+		return envHost
+	}
+	return DefaultHost
+}
+
+// FromEnv creates a new client using Host() as the host. An error is returns
+// if the host is invalid.
```
Comment (justinmayer) [inline api/client.go:54]: Typo: `returns` → `returned`

**Review by justinmayer — CHANGES_REQUESTED** [review]
I made a few minor suggestions. In addition, I don't think this should be merged until accompanying documentation is added, preferably to the README since the `docs` directory isn't linked anywhere and is thus very hard to discover at the moment.

**Comment by bmizerany** [comment]
> I don't think this should be merged until accompanying documentation is added, preferably to the README

@justinmayer Where in the README should I add it to? I don't see any mention of environment variables in it ATM? Open to me picking a spot?

**Comment by drhino** [comment]
in #271 i've added it below the rest api documention.
I also noticed that technoevangelist is writing an api.md in #289

Can't hurt to have it listed in both!

**Comment by justinmayer** [comment]
I think underneath the REST API section in the README would be a good place to put this documentation.

And as @drhino suggested, perhaps also putting this information in the `api.md` documentation would be a good idea as well. 👍

**Comment by jmorganca** [comment]
@bmizerany thanks for the PR! Great to see a friendly face 😊 .

Sorry for the delay in reviewing/merging. Need to consider the env var naming scheme we want here and will reply soon so we can get this merged!

File: api/client.go [hunk 3]
```diff
@@ -9,10 +9,17 @@ import (
 	"io"
 	"net/http"
 	"net/url"
+	"os"
+)
+
+const DefaultHost = "localhost:11434"
+
+var (
+	envHost = os.Getenv("OLLAMA_CLIENT_HOST")
```
Comment (jmorganca) [inline api/client.go:18]: Possible to update this to `OLLAMA_HOST` (sorry for the switcheroo)

**Comment by jmorganca** [comment]
Looks great! Just circled back with @pdevine and `OLLAMA_HOST` is best – similar to Docker and other tools. Thanks for the PR!!

File: api/client.go (hunk 3 above)
Comment (jmorganca) [inline api/client.go:18]:
```suggestion
	envHost = os.Getenv("OLLAMA_HOST")
```

File: api/client.go [hunk 4]
```diff
@@ -33,14 +40,34 @@ func checkError(resp *http.Response, body []byte) error {
 	return apiError
 }
 
+// Host returns the default host to use for the client. It is determined in the following order:
+// 1. The OLLAMA_CLIENT_HOST environment variable
```
Comment (jmorganca) [inline api/client.go:44]:
```suggestion
// 1. The OLLAMA_HOST environment variable
```

**Review by jmorganca — APPROVED** [review]

**Pull request merged**